### ISSUE TITLE
cyberRT scheduling failed to set RT policy

### ISSUE THREAD

**Issue by byhongda** [ApolloAuto/apollo#12823]
**Describe the bug**

**To Reproduce**
1. modify cyberRT scheduling policy
  vim /apollo/cyber/conf/compute_sched.conf
  replace default processor_policy "SCHED_OTHER" to SCHED_FIFO
2. start a modules by Dreamview, or by command in the docker shell
  mainboard -d /apollo/modules/control/dag/control.dag -p compute_sched
3. use `ps -efc | grep mainboard` to check process's scheduling info
```
apollo 10397 10374 TS 19 16:03 ? 00:01:58 mainboard -d /apollo/modules/control/dag/control.dag -p compute_sched
```

**Expected behavior**
The TS(SCHED_OTHER) above should be FF(SCHED_FIFO)

**Try to fix it**
I tried to debug this issue, and it seems like a permission problem.
https://github.com/ApolloAuto/apollo/blob/d0133867997724948a846baf3d236f4db08a72dc/cyber/scheduler/common/pin_thread.cc#L85
The code above doesn't check the return value of pthread_setschedparam() function.
I print the return value and found that pthread_setschedparam return -1 (EPERM)

I also tried the following:
setcap cap_sys_nice+ep /apollo/bazel-bin/cyber/mainboard
But it says:
```
mainboard: error while loading shared libraries: libcyber_Slibcyber_Ucore.so: cannot open shared object file: No such file or directory
```

So, any other good solutions?

**Comment by daohu527** [comment]
The man [pthread_setschedparam](https://man7.org/linux/man-pages/man3/pthread_setschedparam.3.html) already have an example, I think maybe other problems?

**Comment by byhongda** [comment]
> The man [pthread_setschedparam](https://man7.org/linux/man-pages/man3/pthread_setschedparam.3.html) already have an example, I think maybe other problems?

Yes, the docker runtime permission problem.

**Comment by daohu527** [comment]
You're right. Only **CAP_SYS_NICE** level threads can modify the priority, the [sched ](https://man7.org/linux/man-pages/man7/sched.7.html)doc below.
> Privileges and resource limits
In Linux kernels before 2.6.12, only privileged (CAP_SYS_NICE)
       threads can set a nonzero static priority (i.e., set a real-time
       scheduling policy).  The only change that an unprivileged thread can
       make is to set the SCHED_OTHER policy, and this can be done only if
       the effective user ID of the caller matches the real or effective
       user ID of the target thread (i.e., the thread specified by pid)
       whose policy is being changed.

**Comment by byhongda** [comment]
@daohu527  So, the conclusion is that the cyber's RT scheduler can only be used under the root privilege.
Even the official docker can not work well for the process-level RT scheduling?

**Comment by daohu527** [comment]
> Since Linux 2.6.12, the RLIMIT_RTPRIO resource limit defines a
       ceiling on an unprivileged thread's static priority for the SCHED_RR
       and SCHED_FIFO policies.  The rules for changing scheduling policy
       and priority are as follows:

I think we need to read the documentation and perform some tests. : )

**Comment by byhongda** [comment]
> > Since Linux 2.6.12, the RLIMIT_RTPRIO resource limit defines a
> > ceiling on an unprivileged thread's static priority for the SCHED_RR
> > and SCHED_FIFO policies.  The rules for changing scheduling policy
> > and priority are as follows:
> 
> I think we need to read the documentation and perform some tests. : )

Maybe you will struggle with Linux's security mechanism. Good luck~
